tests/test_user_panel.py

```diff
@@ -323,39 +344,133 @@ def test_assigning_resets_tickets(db, user_client):
     assert ticket.p3_conference.days == new_tc.days
 
 
-@pytest.mark.xfail
-def test_profile_settings_requires_login():
-    assert False
+def test_profile_settings_requires_login(client):
+    url = reverse('user_panel:profile_settings')
 
+    response = client.get(url)
 
-@pytest.mark.xfail
-def test_profile_settings_gets_initial_data():
-    assert False
+    assert redirects_to(response, reverse("accounts:login"))
 
 
-@pytest.mark.xfail
-def test_profile_settings_updates_user_data():
-    assert False
+def test_profile_settings_gets_initial_data(user_client):
+    url = reverse('user_panel:profile_settings')
+    user = user_client.user
+    attendee_profile = user.attendeeprofile
+    p3_profile = attendee_profile.p3_profile
 
+    response = user_client.get(url)
 
-@pytest.mark.xfail
-def test_profile_settings_forbids_using_registered_email():
-    assert False
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/user_panel/profile_settings.html")
+    assert user.first_name in response.content.decode()
+    assert user.last_name in response.content.decode()
+    assert user.email in response.content.decode()
+    assert p3_profile.tagline in response.content.decode()
+    assert p3_profile.twitter in response.content.decode()
+    assert attendee_profile.getBio().body in response.content.decode()
+
+
+def test_profile_settings_updates_user_data(user_client):
+    url = reverse('user_panel:profile_settings')
+    payload = dict(
```

Comment:
nitpick question – why `dict` and not `{}` literal?

Reply:
Less typing :)